src/QuiqChatClient.js

```diff
@@ -2,22 +2,25 @@
 import * as API from './apiCalls';
 import {setGlobals, checkRequiredSettings} from './globals';
 import {connectSocket, disconnectSocket} from './websockets';
-import type {AtmosphereMessage, Message, ApiError, UserEventTypes} from './types';
+import type {AtmosphereMessage, TextMessage, ApiError, UserEventTypes, Event} from './types';
 import {MessageTypes, quiqChatContinuationCookie} from './appConstants';
 import {set, get} from 'js-cookie';
-import {differenceBy, last} from 'lodash';
+import {differenceBy, last, partition} from 'lodash';
+import {sortByTimestamp} from './utils';
 
-const getConversationMessages = async () => {
+const getConversation = async (): Promise<{events: Array<Event>, messages: Array<TextMessage>}> => {
   const conversation = await API.fetchConversation();
-  return conversation.messages.filter(
-    m =>
-      m.type === MessageTypes.TEXT &&
-      !m.text.trim().includes('Quiq Welcome Form Customer Submission'),
+  const partitionedConversation = partition(conversation.messages, {type: MessageTypes.TEXT});
+  // NOTE: For backwards compatibility, we must filter out 'Quiq Welcome Form Customer Submission' textMessages
+  const messages = partitionedConversation[0].filter(
+    m => !m.text.trim().includes('Quiq Welcome Form Customer Submission'),
```

Comment:
Oh, just saw your comment. I still think we should be fine to remove it.  I doubt any of the old webchat users will still be connected for long, and the few that may be would just see their welcome form data as a message